### PR TITLE
Add `streamingChatLanguageModelSupplier` property to `@RegisterAiService`

### LINKED ISSUE #842: Support custom `StreamingChatLanguageModel` suppliers
`RegisterAiService` currently allows adding a `chatLanguageModelSupplier`. Unfortunately, I could not find a straightforward way of implementing a custom `Supplier<StreamingChatLanguageModel>`. 

I'd suggest adding an additional property `streamingChatLanguageModelSupplier` to `RegisterAiService`, so we can use custom models.

I could not find a viable alternative for providing my own implementation, except for writing my own extension (like in `quarkus-langchain4j-azure-openai`), which seems to be overkill for that task. 

I also think `quarkus-langchain4j` would profit from that feature, as it would allow to add more test cases with `StreamingChatLanguageModel`s, when a custom model can be implemented like in: 

```java
public class StreamingTest {

    static class MyModelSupplier implements Supplier<StreamingChatLanguageModel> {
        @Override
        public StreamingChatLanguageModel get() {
            return (messages, handler) -> {
                handler.onComplete(new Response<>(new AiMessage("42")));
            };
        }
    }
    
    @RegisterAiService(streamingChatLanguageModelSupplier = MyModelSupplier.class)
    interface MyService {
        @UserMessage("What is the Answer to the Ultimate Question of Life, the Universe, and Everything?")
        Multi<String> chat();
    }

// ...
}
```

### PR DESCRIPTION
**Pull request by dastrobu**
This is need to support custom `StreamingChatLanguageModel` in Ai Services

- Closes: #842

### CODE REVIEW

**Comment by geoand** [comment]
Thanks!

I'll review the PR next week!

File: core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java [hunk 1]
```diff
@@ -359,6 +362,20 @@ public void findDeclarativeServices(CombinedIndexBuildItem indexBuildItem,
         }
     }
 
+    private DotName dotName(AnnotationValue instanceAnnotation, DotName supplierDotName, Consumer<DotName> validator) {
```
Comment (geoand) [inline core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java:365]: Can we please have a better name for this method?

File: core/runtime/src/main/java/io/quarkiverse/langchain4j/runtime/AiServicesRecorder.java [hunk 2]
```diff
@@ -252,4 +254,11 @@ public T apply(SyntheticCreationalContext<T> creationalContext) {
             }
         };
     }
+
+    private static <T> Supplier<T> getSupplier(String className) throws InstantiationException, IllegalAccessException,
```
Comment (geoand) [inline core/runtime/src/main/java/io/quarkiverse/langchain4j/runtime/AiServicesRecorder.java:258]:
```suggestion
    private static <T> Supplier<T> createSupplier(String className) throws InstantiationException, IllegalAccessException,
```

**Review by geoand — COMMENTED** [review]
Looks good!

I just added a couple small comments

File: core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java (hunk 1 above)
Comment (dastrobu) [reply core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java]: changed it to `getSupplierDotName`, maybe you have an even better idea?

Comment (geoand) [reply core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java]: Sounds good!

**Review by geoand — APPROVED** [review]

**Pull request merged**